hostinfo/fileperms.lua

```diff
@@ -0,0 +1,84 @@
+--[[
+Copyright 2015 Rackspace
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+]]--
+local table = require('table')
+local los = require('los')
+local async = require('async')
+local fs = require('fs')
+local execFileToBuffers = require('./misc').execFileToBuffers
+
+--[[ Check file permissions ]]--
+local HostInfo = require('./base').HostInfo
+local Info = HostInfo:extend()
+function Info:initialize()
+    HostInfo.initialize(self)
+end
+
+function Info:run(callback)
+    if los.type() ~= 'linux' then
+        self._error = 'Unsupported OS for file permissions'
+        callback()
+        return
+    end
+
+    local fileList = {
+        '/etc/grub.conf',
+        '/boot/grub/grub.cfg',
+        '/etc/passwd',
+        '/etc/shadow',
+        '/etc/hosts.allow',
+        '/etc/hosts.deny',
+        '/etc/anacrontab',
+        '/etc/crontab',
+        '/etc/cron.hourly',
+        '/etc/cron.daily',
+        '/etc/cron.weekly',
+        '/etc/cron.monthly',
+        '/etc/cron.d',
+        '/etc/ssh/sshd_config',
+        '/etc/gshadow',
+        '/etc/group',
+        '/etc/login.defs',
+        '/var/run/php-fpm.sock'
+    }
+
+    async.forEachSeries(fileList, function(file, cb)
+        local args = {'-L', '-c', '%a', file }
+        local cmd = 'stat'
+        local function execCB(err, exitcode, stdout_data, stderr_data)
+            if exitcode ~= 0 then
+                self._error = fmt("Packages exited with a %d exitcode", exitcode)
+                callback()
+                return
+            end
+            local obj = {}
+            obj[file] = tonumber((stdout_data:gsub('"(%d+)"', "%1")))
```

Comment:
Instead of using gsub, it's simpler and faster to just use `:match`.  If you don't have a capture group, it will return the entire match.

```lua
stdout:data:match("%d+")
```

If you really want to only match a number between quotes, use the same pattern as before, but with match:

```lua
stdout_data:match('"(%d+)"')
```